File: backend/scripts/create_test_user.py
#!/usr/bin/env python3
"""
テストユーザーを作成するスクリプト
使用方法: python3 create_test_user.py <email> <password>
"""

import sys
import sqlite3
import bcrypt
from datetime import datetime

if len(sys.argv) != 3:
    print("使用方法: python3 create_test_user.py <email> <password>")
    print("例: python3 create_test_user.py test@example.com password123")
    sys.exit(1)

email = sys.argv[1]
password = sys.argv[2]

# パスワードをハッシュ化
password_hash = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')

# データベースに接続
conn = sqlite3.connect('data/app.db')
cursor = conn.cursor()

# 既存のユーザーをチェック
cursor.execute("SELECT id FROM users WHERE email = ?", (email,))
if cursor.fetchone():
    print(f"エラー: ユーザー {email} は既に存在します")
    conn.close()
    sys.exit(1)

# ユーザーを作成（メール確認済み）
now = datetime.now().isoformat()
cursor.execute(
    """INSERT INTO users (email, password_hash, email_verified, created_at, updated_at)
       VALUES (?, ?, 1, ?, ?)""",
    (email, password_hash, now, now)
)

conn.commit()
conn.close()

print(f"ユーザーを作成しました: {email}")
print("メール確認済み（email_verified = 1）として設定しました")
print("このユーザーでログインできます。")

File: backend/scripts/create_test_user.rs
// テストユーザーを作成するRustスクリプト
// コンパイル: rustc --edition 2021 create_test_user.rs -L target/debug/deps --extern bcrypt --extern sqlx
// 実行: ./create_test_user <email> <password>

use std::env;
use std::process;

fn main() {
    let args: Vec<String> = env::args().collect();
    
    if args.len() != 3 {
        eprintln!("使用方法: {} <email> <password>", args[0]);
        eprintln!("例: {} test@example.com password123", args[0]);
        process::exit(1);
    }
    
    let email = &args[1];
    let password = &args[2];
    
    // パスワードをハッシュ化
    let password_hash = bcrypt::hash(password, bcrypt::DEFAULT_COST)
        .expect("Failed to hash password");
    
    // データベースに接続
    let db_url = "sqlite://data/app.db";
    // 実際の実装では、sqlxを使用してデータベースに接続する必要があります
    // ここでは簡易版として、SQLコマンドを出力します
    
    println!("以下のSQLコマンドを実行してください:");
    println!("sqlite3 data/app.db \"INSERT INTO users (email, password_hash, email_verified, created_at, updated_at) VALUES ('{}', '{}', 1, datetime('now'), datetime('now'));\"", email, password_hash);
}

